frontend/public/components/utils/managed-by.tsx

```diff
@@ -0,0 +1,116 @@
+import {
+  K8sResourceCommon,
+  K8sResourceKind,
+  referenceForGroupVersionKind,
+  referenceForModel,
+  referenceForOwnerRef,
+  OwnerReference,
+} from '../../module/k8s';
+import { RootState } from '@console/internal/redux';
+import { withFallback } from '@console/shared/src/components/error/error-boundary';
+import { ClusterServiceVersionModel } from '@console/operator-lifecycle-manager/src/models';
+import { ClusterServiceVersionKind } from '@console/operator-lifecycle-manager';
+import { k8sList } from '../../module/k8s/resource';
+
+const managedByLinkStateToProps = (state: RootState): managedByLinkStateProps => {
+  return {
+    activeNamespace: state.UI.get('activeNamespace'),
+  };
+};
+
+const isOwnedByOperator = (csv: K8sResourceKind, owner: OwnerReference) => {
+  return (
+    csv.spec.customresourcedefinitions.owned.find((api) => api.kind === owner.kind) ===
+    csv.spec.customresourcedefinitions.owned.find((api) => owner.apiVersion === api.version)
+  );
+};
+
+const matchOwnerAndCSV = (owner: OwnerReference, csvs: K8sResourceCommon[] = []) => {
+  return csvs.find((csv) => isOwnedByOperator(csv, owner));
+};
+
+const ManagedByOperatorResourceLink: React.SFC<ManagerLinkProps> = (props) => {
+  const { csvName, group, kind, namespace, name, owner, version } = props;
+
+  const reference = referenceForGroupVersionKind(group)(version)(kind);
+
+  const link = `/k8s/ns/${namespace}/${referenceForModel(
+    ClusterServiceVersionModel,
+  )}/${csvName}/${reference}/${name}`;
+
+  return (
+    <span className="co-resource-item">
+      <ResourceIcon kind={referenceForOwnerRef(owner)} />
+      <Link to={link} className="co-resource-item__resource-name" data-test-operand-link={name}>
+        {name}
+      </Link>
+    </span>
+  );
+};
+
+const namespacedManagedByOperatorLink = withFallback<managedByLinkProps>(
+  ({ activeNamespace, className, obj }) => {
+    const [data, setData] = React.useState<ClusterServiceVersionKind[]>();
+    React.useEffect(() => {
+      k8sList(ClusterServiceVersionModel, { ns: activeNamespace })
+        .then(setData)
+        .catch(
+          // eslint-disable-next-line no-console
+          console.error('Could not fetch CSVs'),
+        );
+    }, [activeNamespace]);
+    const ownerReferences = obj?.metadata?.ownerReferences || [];
+    const owner = data
+      ? ownerReferences.find((o) => data.some((csv) => isOwnedByOperator(csv, o)))
+      : undefined;
+    const csv = data && owner ? matchOwnerAndCSV(owner, data) : undefined;
+
+    if (owner && csv) {
+      return (
+        <div className={classNames('co-m-pane__heading-owner', className)}>
+          Managed by{' '}
+          <ManagedByOperatorResourceLink
+            namespace={activeNamespace}
+            csvName={csv.metadata.name}
+            name={owner.name}
+            owner={owner}
+            group={owner.apiVersion.split('/')[0]}
+            version={owner.apiVersion.split('/')[1]}
```

Comment:
This looks like it will be wrong for core `v1` resources that don't have a `/` in the `apiVersion`. I'd suggest using `groupVersionFor`.